staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go

```diff
@@ -235,6 +235,16 @@ type ValidationRule struct {
 	// If unset, the message is "failed rule: {Rule}".
 	// e.g. "must be a URL with the host matching spec.host"
 	Message string `json:"message,omitempty" protobuf:"bytes,2,opt,name=message"`
+	// MessageExpression declares a CEL expression that evaluates to the validation failure message that is returned when this rule fails.
+	// Since MessageExpression is used as a failure message, it must evaluate to a string at compilation time.
```

Comment:
nit: remove `at compilation time` since user should not care about the concept of compilation?

Reply:
Good point, I'll remove that.